### ISSUE TITLE
server.remove is called for all `local` files on destroy()

### ISSUE THREAD

**Issue by albertjandev** [pqina/filepond#293]
### Summary
If server.remove is set, filepond is sending remove request for files with `local` origin when it is being destroyed.

I wanted to create a file uploading form, which saves it's status, so when user reopens the form later, users can see already uploaded files, and add or remove.
So when user reopens form, I have added already uploaded files (got list from another endpoint) as `local` files.

It was working great with 4.2.0.

Then after upgrading to 4.3.x, filepond sends out remove request for all `local` files when it is being destroyed, making already uploaded files to be deleted.

I digged into the source code, found that on destroy(), when server.remove is set, sending all remove request for `local` files.

### How to reproduce
Install filepond 4.3.x, react-filepond

```js
  componentDidMount() {
    getBpmnImageFileList(this.props.mapId)
      .then((res) => {
        this.setState({
          imageFiles: res.data.files.map(file => ({
            source: file,
            options: { type: 'local' },
          }))
        });
      });
  }

...
// render

<FilePond
  ref={(ref) => {
    this.imagesFilePond = ref;
  }}
  files={imageFiles}
  allowMultiple
  maxFiles={null}
  allowDrop
  allowBrowse
  allowRevert
  dropOnPage={false}
  instantUpload
  labelIdle={'Drag and Drop images, or (<span class="filepond--label-action">Browse</span>)'}
  server={{
    url: `/api/maps/${mapId}/images`,
    process: {
      url: '/',
      method: 'POST',
      headers: {
        'x-access-token': localStorage.token,
      },
    },
    load: {
      url: '/',
      method: 'GET',
      headers: {
        'x-access-token': localStorage.token,
      },
    },
    revert: this.handleImagesFilePondRemoveFile,
    remove: this.handleImagesFilePondRemoveFile,
  }}
  onupdatefiles={this.handleImagesFilePondUpdateFiles}
/>

```

### Expected behaviour
It should not send remove request for already uploaded files on destroy.

### Additional information

...

| Environment  | Version
|------------- | -----------
| OS           | MacOS
| Device       | iMac
| Browser      | Chrome 73

**Comment by rikschennink** [comment]
Thanks for the thorough issue report, will probably be able to fix this next week.

**Comment by rikschennink** [comment]
Just published version `4.3.6` which solves the issue.
